### PR TITLE
[SPARK-33208][SQL] Update the document of SparkSession#sql

### PR DESCRIPTION
**Pull request by waitinfuture**
Change-Id: I82db1f9e8f667573aa3a03e05152cbed0ea7686b

### What changes were proposed in this pull request?
Update the document of SparkSession#sql, mention that this API eagerly runs DDL/DML commands, but not for SELECT queries.

### Why are the changes needed?
To clarify the behavior of SparkSession#sql.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
No needed.

### CODE REVIEW

**Comment by leanken** [comment]
@cloud-fan

**Comment by cloud-fan** [comment]
ok to test

File: sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala [hunk 1]
```diff
@@ -601,6 +601,7 @@ class SparkSession private(
   /**
    * Executes a SQL query using Spark, returning the result as a `DataFrame`.
    * The dialect that is used for SQL parsing can be configured with 'spark.sql.dialect'.
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala:603]: I don't think we have this config anymore. Can you find out the PR that added this config, and see if there are any other places that we need to update documents?

Comment (leanken) [reply sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala:603]: It seems these comment with 'spark.sql.dialect' was added in "[SPARK-13643][SQL] Implement SparkSession".
the other reference is SQLContext.scala. FYI.

Comment (waitinfuture) [reply sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala:603]: [SPARK-28997] added this config, and [SPARK-30125] removed it. For now SparkSession.sql and SQLContext.sql's comments refer to it. I think we just need to delete it?

Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala:603]: Yes, let's remove all the references of `spark.sql.dialect`

Comment (waitinfuture) [reply sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala]: done

File: sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala [hunk 2]
```diff
@@ -641,8 +641,8 @@ class SQLContext private[sql](val sparkSession: SparkSession)
   }
 
   /**
-   * Executes a SQL query using Spark, returning the result as a `DataFrame`. The dialect that is
-   * used for SQL parsing can be configured with 'spark.sql.dialect'.
+   * Executes a SQL query using Spark, returning the result as a `DataFrame`.
+   * The dialect that is used for SQL parsing can be configured with 'spark.sql.dialect'.
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala:645]: Let's remove it.

Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala:645]: the doc here should be the same as `SparkSession.sql`

Comment (waitinfuture) [reply sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala]: Oh, that was a mistake. Fixed.

**Comment by cloud-fan** [comment]
thanks, merging to master/3.0!

**Comment by cloud-fan** [comment]
@waitinfuture can you leave a comment in the JIRA so that I can assign the ticket to you?

**Comment by waitinfuture** [comment]
> @waitinfuture can you leave a comment in the JIRA so that I can assign the ticket to you?

done

**Pull request closed without merging**